### PR TITLE
Utilization capped to one

### PR DESCRIPTION
**Pull request by 0xjocke**
Also adds a utilization digest

### CODE REVIEW

File: markets/bfp-market/contracts/modules/PerpMarketFactoryModule.sol [hunk 1]
```diff
@@ -162,6 +162,22 @@ contract PerpMarketFactoryModule is IPerpMarketFactoryModule {
         return PerpMarket.load().activeMarketIds;
     }
 
+    function getUtilizationDigest(
```
Comment (0xjocke) [inline markets/bfp-market/contracts/modules/PerpMarketFactoryModule.sol:165]:
```suggestion
    /// @inheritdoc IPerpMarketFactoryModule
    function getUtilizationDigest(
```

File: markets/bfp-market/contracts/interfaces/IPerpMarketFactoryModule.sol [hunk 2]
```diff
@@ -23,6 +23,17 @@ interface IPerpMarketFactoryModule is IMarket, IBasePerpMarket {
...
     }
 
+    struct UtilizationDigest {
+        /// Last computed utilization rate.
+        uint256 lastComputedUtilizationRate;
+        /// Timestamp of last computed utilization rate.
+        uint256 lastComputedTimestamp;
+        /// Utilization rate, will become active is anyone call recomputeUtilization.
+        uint256 currentUtilizationRate;
+        /// Current utilisation. Will be used to calculate the utilization rate is anyone call recomputeUtilization.
```
Comment (davidvuong) [inline markets/bfp-market/contracts/interfaces/IPerpMarketFactoryModule.sol:33]:
```suggestion
        /// The current instantaneous collateral utilization.
```

File: markets/bfp-market/contracts/interfaces/IPerpMarketFactoryModule.sol [hunk 3]
```diff
@@ -97,6 +108,11 @@ interface IPerpMarketFactoryModule is IMarket, IBasePerpMarket {
         uint128 marketId
     ) external view returns (IPerpMarketFactoryModule.MarketDigest memory);
 
+    /// @notice Returns a digest of utilisation data.
```
Comment (davidvuong) [inline markets/bfp-market/contracts/interfaces/IPerpMarketFactoryModule.sol:111]:
```suggestion
    /// @notice Returns a the utilization digest of an existing market given their `marketId`.
    /// @param marketId Market to query the digest against
    /// @return getUtilizationDigest Utilization digest struct
```

File: markets/bfp-market/test/integration/modules/PerpMarketFactoryModule.utilisation.test.ts [hunk 4]
```diff
@@ -137,16 +140,81 @@ describe('PerpMarketFactoryModule Utilization', () => {
...
       const { BfpMarketProxy, Core } = systems();
 
       const market = genOneOf(markets());
       const marketId = market.marketId();
 
-      const globalMarketConfig = await BfpMarketProxy.getMarketConfiguration();
-      const utilizationBreakpointPercent = wei(globalMarketConfig.utilizationBreakpointPercent);
-      const lowUtilizationSlopePercent = wei(globalMarketConfig.lowUtilizationSlopePercent);
-      const highUtilizationSlopePercent = wei(globalMarketConfig.highUtilizationSlopePercent);
+      // Change staking to the minimum about
```
Comment (davidvuong) [inline markets/bfp-market/test/integration/modules/PerpMarketFactoryModule.utilisation.test.ts:149]: amount?

**Review by davidvuong — APPROVED** [review]
Left some minors comments

**Comment by davidvuong** [comment]
Tests failing due to:

```
[synthetix-v3]: [@synthetixio/buyback-snx]: Executing [provision.oracle_manager]...
[synthetix-v3]: [@synthetixio/buyback-snx]:   Resolving oracle-manager:latest@main (Chain ID: 13370) via local, 0x8E5C7EFC9636A6A0408A46BB7F617094B81e5dba...
[synthetix-v3]: [@synthetixio/buyback-snx]:     Downloading ipfs://QmfDgnQyUNCkKg9amftCFiE4AyPt933dbCkU3rAGWDJ5mV via ipfs https://us-east.repo.usecannon.com
[synthetix-v3]: [@synthetixio/buyback-snx]: An unexpected error occurred:
[synthetix-v3]: [@synthetixio/buyback-snx]: 
[synthetix-v3]: [@synthetixio/buyback-snx]: Error: could not download cannon package data from "QmfDgnQyUNCkKg9amftCFiE4AyPt933dbCkU3rAGWDJ5mV": AxiosError: Request failed with status code 403
```

🤡

**Pull request merged**